lib/core/src/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobin.hs

```diff
@@ -855,11 +870,25 @@ makeChange minCoinFor requiredCost mExtraCoinSource inputBundles outputBundles
         -- Next, sort the list into ascending order of asset count, which moves
         -- any empty maps to the start of the list:
         & NE.sortWith (AssetCount . fst)
-        -- Finally, combine the existing list with the change maps for non-user
+        -- Next, combine the existing list with the change maps for non-user
         -- specified assets, which are already sorted into ascending order of
         -- asset count:
         & NE.zipWith (\m1 (m2, c) -> (m1 <> m2, c))
             changeForNonUserSpecifiedAssets
+        -- Finally, if there are any maps with excessive token quantities, then
+        -- split these maps up along with their corresponding output coins:
+        & splitMapsWithExcessiveQuantities
+      where
+        splitMapsWithExcessiveQuantities
+            :: NonEmpty (TokenMap, Coin) -> NonEmpty (TokenMap, Coin)
+        splitMapsWithExcessiveQuantities =
```

Comment:
I would prefer if the unit testing were based on properties of this function.

Reply:
> I would prefer if the unit testing were based on properties of this function.

Fair enough.

Since the critical part of this function relates to list-processing, I've extracted this out as a separate function, for which I've added some simple property tests.

Now the `splitMapsWithExcessiveQuantities` function is simply a wrapper function that bundles and unbundles the `(TokenMap, Coin)` pair.

See `9cf72cba44204032a1238c40e4c7ee156ad6914d`.